homeassistant/components/template/sensor.py

```diff
@@ -145,6 +147,18 @@ async def async_added_to_hass(self):
         if self._friendly_name_template is not None:
             self.add_template_attribute("_name", self._friendly_name_template)
 
+        # restore state after startup recovering number if possible
+        state = await self.async_get_last_state()
```

Comment:
I don't think that this makes sense. Templates are based on the current state in the state machine. Restoring the state means it's no longer based on the current state?

Reply:
I completely understand your comment, therefore my initial open question about this PR on Discord.

The point for me is that a template value is immediately available on hass start based on his last_state.

Reply:
This one is a bit of a head scratcher for me.  Keeping the old state on restart means that automations that are doing something like from `off` to `on` for `00:02:00` keep working though a restart as the user usually isn't expecting it to go to an `unknown` state between when template is loaded and the start went.   That seems fine for most cases since it is similar to the state being missed between device polls.  It might be a problem for the case the restart takes a while or Home Assistant is stopped for a while.